src/main/daml/Daml/Finance/Instrument/Swap/Fpml/Util.daml

```diff
@@ -544,6 +558,36 @@ calculateFloatingPaymentClaimsFromSwapStream floatingRateCalculation s
             where
               rate = getFloatingRate s pp.calculationPeriod p regularRate rateFixingCalendar fixLeg
 
+      -- calculate simple interest rate claim (no calc period compounding) for a payment period
+      -- using a compounded reference rate
+      createSimpleClaimCompoundedRefRate regularRate p =
+        mconcat $ map createSimpleRateCompoundingRefRate paymentPeriodData
+        where
+          (notional, paymentPeriodData) = calculateNotionalAndFixings p s
+            calculationScheduleStaticData fxRateId useAdjustedDatesForDcf periodicSchedule
+          createSimpleRateCompoundingRefRate pp = mconcat $ [claim, correctionClaim]
+            where
+              rate = getFloatingRate s pp.calculationPeriod p regularRate rateFixingCalendar fixLeg
+                -- Calculate dcf using the method implicitly used for the rate Index,
+                -- for example Act360 in the case of SOFR Index, see page 35 here:
+                -- https://www.newyorkfed.org/medialibrary/Microsites/arrc/files/2021/users-guide-to-sofr2021-update.pdf
+              dcfImpl = Const (calcPeriodDcf (fromSome dcfConventionImplied)
+                pp.calculationPeriod useAdjustedDatesForDcf periodicSchedule.terminationDate
+                periodicSchedule.frequency)
+              notionalAndDcf = if pp.rateFixingDate < pp.calculationPeriod.adjustedEndDate then
+                error "fixing of notional must not occur before the final calculation date"
+                else
+                  when (TimeGte pp.rateFixingDate) $ scale notional $
```

Comment:
Do we need this `When`?

Reply:
Yes, sometimes the notional is FX rate adjusted:
```
    notional = case fxRateId of
      None -> notionalConst
      Some fxRateId -> if fxAdjRequired then notionalConst * fxRateObs else notionalConst
        where fxRateObs = Observe fxRateId
```
of course, with `ObserveAt` this (and other) problems would go away. :)